R/utils.R

```diff
@@ -303,6 +285,55 @@ get_model_directory <- function() {
   return(.mod_dir)
 }
 
+#' Set global bbi_exe_path option
+#'
+#' Sets `options('rbabylon.bbi_exe_path')` to the absolute path passed to `.path`.
+#' Note that the file must exist or this will error.
+#' This is used whenever a function calls out to `bbi` to execute something, for example in `submit_model()` and `model_summary()`.
+#' @param .path Path, either from working directory or absolute, that will be set as `options('rbabylon.bbi_exe_path')`
+#' @rdname set_bbi_exe_path
+#' @export
+set_bbi_exe_path <- function(.path) {
+  if (is.null(.path)) {
+    options('rbabylon.bbi_exe_path' = NULL)
+  } else {
+    options('rbabylon.bbi_exe_path' = normalizePath(.path, mustWork = TRUE))
+  }
+
+  cat(glue("options('rbabylon.bbi_exe_path') set to {options('rbabylon.bbi_exe_path')}"))
+}
+
+
+#' Get global bbi_exe_path option
+#'
+#' Gets the path set to `options('rbabylon.bbi_exe_path')` and checks that it is both absolute and exists.
+#' This is used whenever a function calls out to `bbi` to execute something, for example in `submit_model()` and `model_summary()`.
+#' @rdname set_bbi_exe_path
+#' @importFrom fs is_absolute_path file_exists
+#' @export
+get_bbi_exe_path <- function() {
+  .bbi_path <- getOption("rbabylon.bbi_exe_path")
+  if (is.null(.bbi_path)) {
+    return(NULL)
+  }
+
+  if (!fs::is_absolute_path(.bbi_path)) {
```

Comment:
I'm feeling like we want a generic helper for `check_if_exists_and_is_absolute()`. The tricky part is throwing useful errors out of it. I'll think on that...